staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -968,6 +968,7 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericapifilters.WithMuxAndDiscoveryComplete(handler, c.lifecycleSignals.MuxAndDiscoveryComplete.Signaled())
 	handler = genericfilters.WithPanicRecovery(handler, c.RequestInfoResolver)
 	handler = genericapifilters.WithAuditInit(handler)
+	handler = genericfilters.WithRoutine(handler, c.LongRunningFunc, c.RequestInfoResolver)
```

Comment:
I understand the idea, but that seems risky to me.
In general `WithPanicRecovery` is supposed to protect us from panics, so by putting something earlier in the chain, we are risking that it will not play its role correctly...
@liggitt - FYI